### PR TITLE
CHORE: Add Status Codes to All Routes CU-39p06he

### PR DESCRIPTION
**Pull request by drewlacob**
- add status codes to all routes
- ensures we always exit a route once done since res.json() does not
- 400 or 500 used for failures, unsure if we should uniformly make this
  one or the other (400 bad request, 500 internal server error)

### CODE REVIEW

**Comment by freedom-rains** [comment]
Task linked: [CU-39p06he Response Statuses - BE](https://app.clickup.com/t/39p06he)

**Review by AlamTahmidul — APPROVED** [review]

**Comment by RealDavidH** [comment]
> * add status codes to all routes
> 
>     * ensures we always exit a route once done since res.json() does not
> 
>     * 400 or 500 used for failures, unsure if we should uniformly make this
>       one or the other (400 bad request, 500 internal server error)

Please make it either 400 or 500 for bad request. Just make a choice and then set them all to that, to make it less confusing.

File: server/controllers/userControllers/user.controller.js [hunk 1]
```diff
@@ -9,31 +9,31 @@ module.exports.getAllUser = (req, res) => {
...
+    .then((user) => res.status(200).json(user))
+    .catch((err) => res.status(500).json(err));
 };
 
 //Login user
 module.exports.loginUser = async (req, res) => {
   const checkValidate = validateInput(req.body); //I'll have to make a separate validation function for login
   if (checkValidate.length > 0) {
-    return checkValidate;
+    return res.status(400).json(checkValidate);
```
Comment (freedom-rains) [inline server/controllers/userControllers/user.controller.js:36]: This one still needs to be changed to 500

File: server/controllers/userControllers/user.controller.js [hunk 2]
```diff
@@ -52,16 +52,16 @@ module.exports.loginUser = async (req, res) => {
...
-    .catch((err) => res.json(err));
+    .catch((err) => res.status(500).json(err));
 };
 
 //Create a user
 module.exports.addUser = async (req, res) => {
   const checkValidate = validateInput(req.body);
   if (checkValidate.length > 0) {
-    return checkValidate;
+    return res.status(400).json(checkValidate);
```
Comment (freedom-rains) [inline server/controllers/userControllers/user.controller.js:64]: Change to 500

File: server/controllers/userControllers/user.controller.js [hunk 3]
```diff
@@ -82,15 +82,15 @@ module.exports.addUser = async (req, res) => {
...
+    .then((user) => res.status(200).json(user[0]))
+    .catch((err) => res.status(500).json(err));
 };
 
 //Update a user
 module.exports.updateUser = (req, res) => {
   const checkValidate = validateInput(req.body);
   if (checkValidate.length > 0) {
-    return checkValidate;
+    return res.status(400).json(checkValidate);
```
Comment (freedom-rains) [inline server/controllers/userControllers/user.controller.js:93]: Change to 500

**Review by freedom-rains — CHANGES_REQUESTED** [review]

**Review by freedom-rains — APPROVED** [review]

**Pull request merged**